### ISSUE TITLE
v3 does not list next@13 as a peerDependency but uses its features

### ISSUE THREAD

**Issue by robertwbradford** [auth0/nextjs-auth0#1393]
### Checklist

- [X] The issue can be reproduced in the [nextjs-auth0 sample app](https://github.com/auth0-samples/auth0-nextjs-samples/tree/main/Sample-01) (or N/A).
- [X] I have looked into the [Readme](https://github.com/auth0/nextjs-auth0#readme), [Examples](https://github.com/auth0/nextjs-auth0/blob/main/EXAMPLES.md), and [FAQ](https://github.com/auth0/nextjs-auth0/blob/main/FAQ.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://auth0.github.io/nextjs-auth0/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/nextjs-auth0/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

Hello, I just tried installing v3.1.0 in an existing Next 12 application and am seeing the following when loading a page which requires authentication:

```
Error: Cannot find module 'next/headers'
Require stack:
- /my-app/node_modules/@auth0/nextjs-auth0/dist/http/auth0-next-request-cookies.js
- /my-app/node_modules/@auth0/nextjs-auth0/dist/http/index.js
- /my-app/node_modules/@auth0/nextjs-auth0/dist/session/cache.js
- /my-app/node_modules/@auth0/nextjs-auth0/dist/session/index.js
- /my-app/node_modules/@auth0/nextjs-auth0/dist/shared.js
- /my-app/node_modules/@auth0/nextjs-auth0/dist/index.js
- /my-app/.next/server/pages/api/auth/[...auth0].js
- /my-app/node_modules/next/dist/server/next-server.js
- /my-app/node_modules/next/dist/server/next.js
- /my-app/node_modules/next/dist/server/lib/start-server.js
- /my-app/node_modules/next/dist/cli/next-dev.js
- /my-app/node_modules/next/dist/lib/commands.js
- /my-app/node_modules/next/dist/bin/next
```

It looks like `@auth0/nextjs-auth0@3.0.0` started using the `next/headers` feature (e.g., see https://github.com/auth0/nextjs-auth0/commit/4ef676b1c8c65a0fac459a91472d3ae3e4519cdb#diff-5611749eb12286ec92350b836396c951b6ecd180753a88b540a16f761b5cb38fR1) which became available in next@13. However, the `@auth0/nextjs-auth0` package.json still [lists `"next": ">=10"` as a peer dependency](https://github.com/auth0/nextjs-auth0/blob/v3.1.0/package.json#L135).

Does `@auth0/nextjs-auth0` need a patch to update its peerDeps to list next@13?

### Reproduction

I created https://github.com/robertwbradford/next-12-with-nextjs-auth0-3 to show the issue and found that 'next/navigation' is also referenced but was also not available before next@13. If you clone and run `npm i` and `npm run dev` you will see the following:

```
error - ./node_modules/@auth0/nextjs-auth0/dist/helpers/with-page-auth-required.js:3:0
Module not found: Can't resolve 'next/navigation'

Import trace for requested module:
./node_modules/@auth0/nextjs-auth0/dist/helpers/index.js
./node_modules/@auth0/nextjs-auth0/dist/edge.js
./middleware.js
```

### nextjs-auth0 version

3+

### Next.js version

12

### Node.js version

18

**Comment by adamjmcgrath** [comment]
Thanks for raising this @robertwbradford 

Theoretically, you should be able to use Next@12 and the pages directory with this SDK - let me take a look at why this is failing.

I suspect we need to make `next/navigation` and `next/headers` dynamic imports that only run in the app directory - [as next-auth does](https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/src/next/index.ts#L75)

**Comment by mikecuello17** [comment]
@robertwbradford Having the same issue with Next 12.3.1.
``` error - Error: Cannot find module 'next/headers' ```

@adamjmcgrath Would you recommend downgrading to `2.7.0` while this gets worked on?

**Comment by adamjmcgrath** [comment]
@robertwbradford - I've merged a PR (which I'll release today or early next week) that changes the Next.js 13 only imports to dynamic imports which will enable you to run the SDK in the pages router on Next 12.

This is not enough though for using middleware on Next 12 though (more info in the PR https://github.com/auth0/nextjs-auth0/pull/1396), for that you will need to follow the new advice I've added to the migration guide: https://github.com/auth0/nextjs-auth0/blob/main/V3_MIGRATION_GUIDE.md#using-middleware-on-next-12-requires-custom-webpack-config

**Comment by adamjmcgrath** [comment]
Hi @matan-darrow - apologies for the delay, will do a release this week

**Comment by matan-darrow** [comment]
@adamjmcgrath thanks

**Comment by robertwbradford** [comment]
Hello, I see that this merged but was wondering when the next release is expected. Thanks!

**Comment by adamjmcgrath** [comment]
Hi @robertwbradford - this fix should be in [3.2.0](https://github.com/auth0/nextjs-auth0/releases/tag/v3.2.0)

Are you still seeing an issue?

**Comment by robertwbradford** [comment]
@adamjmcgrath I'm sorry for the false alarm. I missed the additional webpack instructions you had mentioned before:

> This is not enough though for using middleware on Next 12 though (more info in the PR #1396), for that you will need to follow the new advice I've added to the migration guide: https://github.com/auth0/nextjs-auth0/blob/main/V3_MIGRATION_GUIDE.md#using-middleware-on-next-12-requires-custom-webpack-config

I added this and am seeing it work. Thank you.
